misc/python/materialize/checks/range.py

```diff
@@ -0,0 +1,110 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+from typing import List
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+
+
+class Range(Check):
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+            > DROP MATERIALIZED VIEW IF EXISTS int_range_view1;
+            > DROP MATERIALIZED VIEW IF EXISTS int_range_view2;
+            > DROP TABLE IF EXISTS int_range_table;
+            > CREATE TABLE int_range_table (f1 INT, f2 INT8RANGE);
+            > INSERT INTO int_range_table VALUES (1, '[2,16]'::INT8RANGE);
+            > INSERT INTO int_range_table VALUES (2, NULL);
+            """
+            )
+        )
+
+    def manipulate(self) -> List[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                """
+                > CREATE MATERIALIZED VIEW int_range_view1 AS SELECT f1, f2 FROM int_range_table;
```

Comment:
In addition to make sure data of type `RANGE` inserted before remains readable at the end of the test, the other major idea of this test this test is to get interesting stuff to be written in the catalog , where it will then be subject to serializations, upgrades, deserializations and whatnot. For example, if you do `SHOW CREATE MATERIALIZED VIEW` on that view, you will see that its definition is nothing to write home about.

Instead, the materialized views should contain references to casts, RANGE literals and the interesting SQL functions/operators that pertain to ranges. This way, if there is a bug in serializing/deserializing/upgrading any of those instances of the function/operator, the view will get borked one way or another and the test is likely to fail.

If all the complexity is in the materialized view, then in the `validate()` section below you can use simple `SELECT *` queries to validate the output from the view.

For an example, see `numeric_types.py` where the view is quite hairy but the validation SELECT is simple.

Of course, given the multiple RANGE types, operators, functions and whatnot, there is plenty of potential for a combinatorial explosion if one tries to stuff all combinations of stuff in a single view. Make the definition only as complex as you can keep in your head or fit in your terminal.